src/behavioral/chain-of-resp/chain-of-resp-es6.js

```diff
@@ -0,0 +1,68 @@
+class ShoppingCart {
+    constructor() {
+        this.products = [];
+    }
+
+    addProduct(p) {
+        this.products.push(p);
+    };
+}
+
+class Discount {
+    calc(products) {
+        let ndiscount = new NumberDiscount();
+        let pdiscount = new PriceDiscount();
+        let none = new NoneDiscount();
+        ndiscount.setNext(pdiscount);
+        pdiscount.setNext(none);
+        return ndiscount.exec(products);
+    };
+}
+
+class NumberDiscount {
+    constructor() {
+        this.next = null;
+    }
+
+    setNext(fn) {
+        this.next = fn;
+    };
+
+    exec(products) {
+        let result = 0;
+        if (products.length > 3)
+            result = 0.05;
+
+        return result + this.next.exec(products);
+    };
+}
+
+class PriceDiscount{
+    constructor() {
+        this.next = null;
+    }
+
+    setNext(fn) {
+        this.next = fn;
+    };
+
+    exec(products) {
+        let result = 0;
+        let total = products.reduce((a, b)=> {
+            return a + b;
+        });
+
+        if (total >= 500)
+            result = 0.1;
+
+        return result + this.next.exec(products);
+    };
+}
+
+class NoneDiscount {
+    exec() {
+        return 0;
+    };
+}
+
+module.exports = [ShoppingCart, Discount];
```

Comment:
you can use es6 `exports`
[example](https://github.com/fbeline/Design-Patterns-JS/blob/master/src/behavioral/strategy/strategy_es6.js#L29)